core/server/main.cpp

```diff
@@ -46,6 +47,10 @@ int main(int a_argc, char **a_argv) {
     string cfg_file;
     bool gen_keys = false;
 
+    // Note: we may want to dynamically choose type at compile time
```

Comment:
I have an idea on how to implement this already if we would like to proceed sooner rather than later:
utilize `underlying_type` and `numeric_limits::max()` to retrieve the type and its respective max that we can use as a sentinel.